### ISSUE TITLE
Results, Final Results

### ISSUE THREAD

**Issue by pfefferminzz** [ludumdare/ludumdare#990]
You should add COMPO or JAM to the Results headline or Final Results subline.

Some users are converting their rank into a percentage based on the number of Compo or Jam participants. You could provide an official conversion like:

Overall: **423** (3.311 average in 47 ratings) **of ???** (Top ??.? %)

**Comment by local-minimum** [comment]
I haven't checked the data from the api, but it should be easy fix. The one thing I wonder about is compared to what? Number of games in category or number of ranked games in category. In this jam the difference would be big.

**Comment by pfefferminzz** [comment]
I don't know. It was my first LDJam. Didn't you have any percentages in the previous jams? I was just just thinking it would be better to provide official percentages instead of waiting until everyone computes percentages based on different numbers. :)

**Comment by Cerno-b** [comment]
That is something I also thought about. Seems like a quite some people like to show their percentage, but a lot of them are getting it wrong. So what is the correct percentage?

Having created the LD38 results histograms over here https://ldjam.com/events/ludum-dare/38/a-perfectly-lovely-adventure/yall-got-some-more-of-them-infographs, I would argue that the only fair assessment is this:

- keep Jam and Compo separately
- percentage is your rank divided by number of **ranked** entries in that category
- make sure you handle the duplicates right.

Counting only the number of ranked entries for the total is important, because otherwise you will be comparing apples with oranges. You will get a much better percentage, but it won't be fair, because the reason other games have not been ranked is not that they are bad, but because they did not fulfill the requirements of getting x votes.

The duplicates should be handled like this:
Assume a toy example of having 6 entries that share only three ranks: three 1st places, two 2nd places, one 3rd place. 
- A: 1st
- B: 1st
- C: 1st
- D: 2nd
- E: 2nd
- F: 3rd

In order to compute the percentage you would have to adapt the rank like this*:
- A: 1st
- B: 1st
- C: 1st
- D: 4th
- E: 4th
- F: 6th

Now using these numbers you can compute the percentage correctly:
- A: (7 - 1)/6 * 100 = 100%
- B: (7 - 1)/6 * 100 = 100%
- C: (7 - 1)/6 * 100 = 100%
- D: (7 - 4)/6 * 100 = 50%
- E: (7 - 4)/6 * 100 = 50%
- F: (7 - 6)/6 * 100 = 16.7%

This could be computed for each category, in which case the divisor (6) may be different based on how many games ranked in that category.

Since this is not straightforward to understand, I think it would be quite helpful to have this calculated on the results page, otherwise too many people will compute it differently.

*) An alternative would be to simply group the ranks and divide each game's rank by the total number of ranks (in this case we have 3 ranks instead of 6 games, so the games will have a score of 100%, 66.6%, 33.3%) but then you won't get your score normalized by the number of participants, but the number of ranks. It's less precise but could be a viable alternative to keep things simpler and easier to understand, especially if you don't have too many duplicates.

What is definitely incorrect is to divide a game's rank by the number of ranked games.

**Comment by zwrawr** [comment]
So your formula is 
```
( num_compo_entries  + 1 - my_position)/(num_compo_entries) * 100%
```

**Comment by Cerno-b** [comment]
Yes, if my_position is computed the way outlined above.

**Comment by zwrawr** [comment]
I disagree. ~1100 people made compo games , so we should give people there rank out of the games that where submitted. Because that's there rank out of the people that participated.

**Comment by Cerno-b** [comment]
And I disagree to that. There were 1104 people who participated, but only 542 people who ranked. The two sets of 542 ranked and 562 unranked participants are disjoint. If you weren't ranked then you cannot compare to someone who was.

How would you compute the correct percentiles then? Would you put all unranked games below the ranked ones? This causes a lot of problems. 

For instance, if you have a ranked 1.6 star game, would that still be in the higher percentile than an unranked 3.7 star game that didn't rank because it only had 10 votes? That would be highly unfair and skewed.

But we can't simply order all games, independent of whether they're ranked or not, because then the highest percentile would be held by a 5 star game with only 4 votes. For exactly this reason do we have the requirement of a minimum amount of votes to be eligible for a rank. What do you do with games that have no votes? Would it be fair to give them a low percentile just because nobody voted or even played the game? Maybe it was pretty good.

I would think that having all games in the percentile would be nice so the creators wouldn't feel left out, but there is no good way of doing that fairly. The same thing goes for putting them in the ranking. There are rules for what needs to be done to qualify and those same rules should hold whether it's for ranking or computing a percentile.

On the other hand, I would argue that it is much more important to inform people clearly right from the beginning that voting and getting votes is crucial to earn a rank so that everyone who wants to be ranked gets a fair shot of getting ranked.

**Comment by zwrawr** [comment]
Fair points, I guess that there isn't a good way to do what i suggested.

**Comment by povrazor** [comment]
Just to clarify, this issue is asking for the ???'s.

> Overall: 423 (3.311 average in 47 ratings) of ??? (Top ??.? %)

**Comment by Cerno-b** [comment]
Precisely.

**Comment by Cerno-b** [comment]
Ah, sorry, this wasn't my issue so I shouldn't state its purpose. I just wanted to point out a way of computing these values in a fair way.

**Comment by rplnt** [comment]
Bump? 

I think having a tooltip on the rank would be enough. E.g. hovering on my place would show a text displaying percentile. No need to change the visible UI should make this easier to push forward?

Agree with whoever said that only ranked games should be included.

**Comment by jacobwinters** [comment]
Would love to see percentile ranks on the site someday. In the meantime, I wrote a script to do the calculation. In case anyone else wants it:

https://jacobwinters.com/ld-percentile-calculator/
